packages/http/src/mocker/HttpMocker.ts

```diff
@@ -33,24 +33,36 @@ class HttpMocker
     const payloadGenerator: PayloadGenerator =
       config && typeof config.mock !== 'boolean' && config.mock.dynamic ? generate : generateStatic;
 
-    return pipe(
-      withLogger(logger => {
-        // setting default values
-        const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
-        config = config || { mock: false, validateRequest: true, validateResponse: true };
-        const mockConfig: IHttpOperationConfig =
-          config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
-
-        if (!mockConfig.mediaTypes && acceptMediaType) {
-          logger.info(`Request contains an accept header: ${acceptMediaType}`);
-          mockConfig.mediaTypes = acceptMediaType.split(',');
-        }
-
-        return mockConfig;
-      }),
-      chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-      chain(result => assembleResponse(result, payloadGenerator)),
-    );
+    const isHead = input.data.method === 'head';
+    const r = get(resource, ['responses', '0'], { code: 200, headers: [] });
+
+    return isHead
+      ? withLogger(logger => {
```

Comment:
I think we can easily find a way to put this block in the one that's inside and have an early return, without having to construct two `withLogger` blocks. Basically the idea would be to defer such check as much as possible. Does that make sense?